processing/src/main/java/org/apache/druid/query/rowsandcols/column/LongArrayColumn.java

```diff
@@ -0,0 +1,201 @@
+import java.util.Arrays;
+
+public class LongArrayColumn implements Column
+{
+  private final long[] vals;
+
+  public LongArrayColumn(
+      long[] vals
+  )
+  {
+    this.vals = vals;
+  }
+
+  @Nonnull
+  @Override
+  public ColumnAccessor toAccessor()
+  {
+    return new MyColumnAccessor();
+  }
+
+  @Nullable
+  @SuppressWarnings("unchecked")
+  @Override
+  public <T> T as(Class<? extends T> clazz)
+  {
+    if (VectorCopier.class.equals(clazz)) {
+      return (T) (VectorCopier) (into, intoStart) -> {
+        if (Integer.MAX_VALUE - vals.length < intoStart) {
+          throw new ISE(
+              "too many rows!!! intoStart[%,d], vals.length[%,d] combine to exceed max_int",
+              intoStart,
+              vals.length
+          );
+        }
+        for (int i = 0; i < vals.length; ++i) {
+          into[intoStart + i] = vals[i];
+        }
+      };
+    }
+    if (ColumnValueSwapper.class.equals(clazz)) {
+      return (T) (ColumnValueSwapper) (lhs, rhs) -> {
+        long tmp = vals[lhs];
+        vals[lhs] = vals[rhs];
+        vals[rhs] = tmp;
+      };
+    }
+    return null;
+  }
+
+  private class MyColumnAccessor implements BinarySearchableAccessor
+  {
+    @Override
+    public ColumnType getType()
+    {
+      return ColumnType.LONG;
+    }
+
+    @Override
+    public int numRows()
+    {
+      return vals.length;
+    }
+
+    @Override
+    public boolean isNull(int rowNum)
+    {
+      return false;
+    }
+
+    @Override
+    public Object getObject(int rowNum)
+    {
+      return vals[rowNum];
+    }
+
+    @Override
+    public double getDouble(int rowNum)
+    {
+      return vals[rowNum];
+    }
+
+    @Override
+    public float getFloat(int rowNum)
+    {
+      return vals[rowNum];
+    }
+
+    @Override
+    public long getLong(int rowNum)
+    {
+      return vals[rowNum];
+    }
+
+    @Override
+    public int getInt(int rowNum)
+    {
+      return (int) vals[rowNum];
+    }
+
+    @Override
+    public int compareRows(int lhsRowNum, int rhsRowNum)
+    {
+      return Long.compare(vals[lhsRowNum], vals[rhsRowNum]);
+    }
+
+
+    @Override
+    public FindResult findNull(int startIndex, int endIndex)
+    {
+      return FindResult.notFound(endIndex);
+    }
+
+    @Override
+    public FindResult findDouble(int startIndex, int endIndex, double val)
+    {
+      return findLong(startIndex, endIndex, (int) val);
+    }
+
+    @Override
+    public FindResult findFloat(int startIndex, int endIndex, float val)
+    {
+      return findLong(startIndex, endIndex, (int) val);
+    }
+
+    @Override
+    public FindResult findLong(int startIndex, int endIndex, long val)
+    {
+      if (vals[startIndex] == val) {
+        int end = startIndex + 1;
+
+        while (end < endIndex && vals[end] == val) {
+          ++end;
+        }
+        return FindResult.found(startIndex, end);
+      }
+
+      int i = Arrays.binarySearch(vals, startIndex, endIndex, val);
```

Comment:
I know this isn't really new since the others do this too, but is there any guarantee that `vals` is sorted?

Reply:
These methods are part of the `BinarySearchableAccessor`, which is fundamentally just saying "if you want to binary search and you know it's safe, use these methods".  Instead of making the accessor itself implement this, it probably would've been better to make it another interface off of the `Column`.  That's my bad really.

Reply:
Fwiw, the whole `BinarySearchableAccessor` is actually only used by an experimental join implementation that exists to exercise the operator stuff.  Your commentary about the oddities are totally valid and belong against that.  It's not really a solidified part of the code base though, so we should perhaps add some javadocs to the interface itself, talk a bit about why it's awkward and when/if we get back to it, we can improve it.

Reply:
The worst-case time complexity of this method is O(n) if all the values from `startIndex` to `endIndex` are the same. It can be improved by doing two binary searches to find the upper and the lower bound. It's the same implementation in other places as well. Is it worth to switch to that instead? If the array will always have mostly distinct values, then this implementation would work better on average.

Reply:
No, none of the "find" stuff is actually being used for anything meaningful.  Ignore it.

Reply:
I'm not sure if it's more likely that we have distinct values here, but currently, this is in line with the other type columns. If we make a call on using multiple searches instead, that should be consistent with the other columns here.

Reply:
@adarshsanjeev Lets java doc the odity for now and move forward? We can take up this work as part of a follow up refactor ?

Reply:
I have added this to the javadoc in BinarySearchableAccessor